mesonbuild/compilers/detect.py

```diff
@@ -659,6 +661,13 @@ def detect_fortran_compiler(env: 'Environment', for_machine: MachineChoice) -> C
     info = env.machines[for_machine]
     cls: T.Type[FortranCompiler]
     for compiler in compilers:
+        # capture help text for possible fallback
+        try:
+            _, help_out, help_err = Popen_safe_logged(compiler + ["--help"], msg='Detecting compiler via')
+        except OSError as e:
+            popen_exceptions[join_args(compiler + ["--help"])] = e
+            help_out = help_err = ""
```

Comment:
@eli-schwartz said in #12306:
> As long as meson can tell the difference between classic flang and new flang by parsing the output of --help / --version [...] we are good to go because that's what we'll base identification off of, not the executable name.

Since I couldn't see anything else using this, I've tentatively added this now. This is because `flang-new --version` outputs
```
flang-new version 17.0.6
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: E:\miniforge\envs\<env_name>\Library\bin
```
-- IOW, nothing llvm-specific (especially once the rename from `flang-new` to `flang` lands) --, while `flang-new --help` outputs
```
OVERVIEW: flang LLVM compiler

USAGE: flang-new.exe [options] file...

OPTIONS:
  [...]
```